validphys2/src/validphys/pineparser.py

```diff
@@ -0,0 +1,316 @@
+    """
+    if metadata.get("apfelcomb") is None:
+        return None
+
+    # Can't pathlib understand double suffixes?
+    operands = [i.name.replace(f".{EXT}", "") for i in gridpaths]
+    ret = {}
+
+    # Check whether we have a normalization active and whether it affects any of the grids
+    if metadata["apfelcomb"].get("normalization") is not None:
+        norm_info = metadata["apfelcomb"]["normalization"]
+        # Now fill the operands that need normalization
+        ret["normalization"] = [norm_info.get(op, 1.0) for op in operands]
+
+    # Check whether the repetition flag is active
+    if metadata["apfelcomb"].get("repetition_flag") is not None:
+        if len(operands) == 1:
+            ret["repetition_flag"] = operands[0] in metadata["apfelcomb"]["repetition_flag"]
+        else:
+            # Just for the sake of it, let's check whether we did something stupid
+            if any(op in metadata["apfelcomb"]["repetition_flag"] for op in operands):
+                raise ValueError(f"The yaml info for {metadata['target_dataset']} is broken")
+
+    # Check whether the dataset has shifts
+    # Note: this only happens for ATLASZPT8TEVMDIST, if that gets fixed we might as well remove it
+    if metadata["apfelcomb"].get("shifts") is not None:
+        shift_info = metadata["apfelcomb"]["shifts"]
+        ret["shifts"] = [shift_info.get(op, 0) for op in operands]
+
+    return ret
+
+
+###########
+
+
+def _pinelumi_to_columns(pine_luminosity, hadronic):
+    """Makes the pineappl luminosity into the column indices of a dataframe
+    These corresponds to the indices of a flat (14x14) matrix for hadronic observables
+    and the non-zero indices of the 14-flavours for DIS
+
+    Parameters
+    ----------
+        pine_luminosity: list(tuple)
+            list with a pair of flavours per channel
+        hadronic: bool
+            flag for hadronic / DIS observables
+
+    Returns
+    -------
+        list(int): list of labels for the columns
+    """
+    from eko.basis_rotation import evol_basis_pids
+    flav_size = len(evol_basis_pids)
+    columns = []
+    if hadronic:
+        for i, j in pine_luminosity:
+            idx = evol_basis_pids.index(i)
+            jdx = evol_basis_pids.index(j)
+            columns.append(flav_size * idx + jdx)
+    else:
+        # The proton might come from both sides
+        try:
+            columns = [evol_basis_pids.index(i) for _, i in pine_luminosity]
+        except ValueError:
+            columns = [evol_basis_pids.index(i) for i, _ in pine_luminosity]
+    return columns
+
+
+def get_yaml_information(yaml_file, theorypath):
+    """Reads the yaml information from a yaml compound file
+
+    Transitional function: the call to "pineko" might be to some other commondata reader
+    that will know how to extract the information from the commondata
+    """
+    # TODO Tell the reader where to look for the pineappl grids
```

Comment:
```suggestion
    # TODO: Tell the reader where to look for the pineappl grids
```

Reply:
Where do you plan to move them after debugging?

Reply:
Hopefully we will have a Theory 200X soonish...

Reply:
In that case. Why keep the fktables around?

Reply:
which ones?